sql-plugin/src/main/scala/com/nvidia/spark/rapids/SchemaUtils.scala

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import java.util.Optional
+
+import scala.collection.mutable.ArrayBuffer
+import scala.language.implicitConversions
+
+import ai.rapids.cudf.{ColumnView, Table}
+import com.nvidia.spark.rapids.RapidsPluginImplicits.AutoCloseableProducingSeq
+import org.apache.orc.TypeDescription
+
+import org.apache.spark.sql.catalyst.parser.CatalystSqlParser
+import org.apache.spark.sql.catalyst.util.CaseInsensitiveMap
+import org.apache.spark.sql.types._
+
+object SchemaUtils extends Arm {
+  /**
+   * Replaces CharType/VarcharType with StringType recursively in the given data type.
+   */
+  private[rapids] def replaceCharVarcharWithString(dt: DataType): DataType = dt match {
+    case ArrayType(et, nullable) =>
+      ArrayType(replaceCharVarcharWithString(et), nullable)
+    case MapType(kt, vt, nullable) =>
+      MapType(replaceCharVarcharWithString(kt), replaceCharVarcharWithString(vt), nullable)
+    case StructType(fields) =>
+      StructType(fields.map { field =>
+        field.copy(dataType = replaceCharVarcharWithString(field.dataType))
+      })
+    case _: CharType => StringType
+    case _: VarcharType => StringType
```

Comment:
VarcharType was introduced by https://issues.apache.org/jira/browse/SPARK-33480 in Spark 3.1.0.  This code is going to need to be updated either so parts of it are under the shim layer, or we don't deal with VarcharType.

Reply:
Nice catch.
CharType(len: Int) is also introduced in Spark 3.1.0. I chose to remove the code, since GPU ORC reader does not support the two types yet. 
I will add shims if needed in the future.